lib/build.js

```diff
@@ -80,11 +80,16 @@ module.exports = async function build (sourceDir, cliOptions = {}) {
   if (options.siteConfig.serviceWorker) {
     console.log('Generating service worker...')
     const wbb = require('workbox-build')
-    wbb.generateSW({
+    await wbb.generateSW({
       swDest: path.resolve(outDir, 'service-worker.js'),
       globDirectory: outDir,
       globPatterns: ['**\/*.{js,css,html,png,jpg,jpeg,gif,svg,woff,woff2,eot,ttf,otf}']
     })
+    await fs.writeFile(
```

Comment:
Can we use the importScripts option instead?

https://developers.google.com/web/tools/workbox/modules/workbox-build#full_generatesw_config

Reply:
Thank you for the suggestion.

I compared it and the current code, I didn't feel that the option improves the build steps much. We need the following steps with the option:

1. Copy the `service-worker/*.js` files to `outDir` manually. (Webpack doesn't handle the files for service worker because Workbox requires the files Webpack generated.)
2. Give the path to `service-worker/*.js` files to the `importScripts` option.

Step 1 is the same step as the current appending step essentially. Plus, there are some concerns:

1. the `importScripts` option makes extra HTTPS requests.
2. Chrome has a bug that overlooks the changes of imported scripts ([chromium#648295](https://bugs.chromium.org/p/chromium/issues/detail?id=648295)). This implies the change of the `service-worker/*.js` files will not trigger to update service worker's caches.